src/datahike/schema_cache.cljc

```diff
@@ -0,0 +1,41 @@
+(ns datahike.schema-cache
```

Comment:
Yes, it is both somewhat trivial and it has global state. A good way to test this would be tests with distributed readers and writers, potentially with generative testing. Given the reasoning below I am fairly confident though that it is at least sound design wise. Basically state changes must be as isolated (single writer) and rare (only in commit!, gc and db management ops) as possible and we need to keep track of non-monotonic operations (gc, db deletion).